lib/iris/tests/integration/experimental/geovista/test_cube_to_poly.py

```diff
@@ -0,0 +1,66 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the BSD license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Integration tests for the `iris.experimental.geovista.cube_to_polydata` function."""
+
+import numpy as np
+
+from iris import load_cube
+from iris.experimental.geovista import cube_to_polydata
+from iris.experimental.ugrid import PARSE_UGRID_ON_LOAD
+from iris.tests import get_data_path
+
+
+def test_integration_2d():
+    file_path = get_data_path(
+        [
+            "NetCDF",
+            "ORCA2",
+            "votemper.nc",
+        ]
+    )
+    with PARSE_UGRID_ON_LOAD.context():
+        global_cube = load_cube(file_path, "votemper")
+
+    polydata = cube_to_polydata(global_cube[0, 1, :])
+
+    assert polydata.GetNumberOfCells() == 26640
+    assert polydata.GetNumberOfPoints() == 26969
+
+
+def test_integration_1d():
+    file_path = get_data_path(
+        [
+            "NetCDF",
+            "global",
+            "xyt",
+            "SMALL_hires_wind_u_for_ipcc4.nc",
+        ]
+    )
+    with PARSE_UGRID_ON_LOAD.context():
+        global_cube = load_cube(file_path)
+
+    polydata = cube_to_polydata(global_cube[0, :])
+
+    assert polydata.GetNumberOfCells() == 51200
+    assert polydata.GetNumberOfPoints() == 51681
+
+
+def test_integration_mesh():
```

Comment:
I've done some rough plotting to demonstrate this is doing the right thing:

<img width="250" alt="Screenshot 2024-03-27 151500" src="https://github.com/SciTools/iris/assets/40734014/df47bb64-783c-48f0-adbe-40efca991a36">